app/actions/views/channel.js

```diff
@@ -906,3 +921,69 @@ function fetchGroupMessageProfilesIfNeeded(state, channel, channelMembers, profi
 
     return null;
 }
+
+export function loadUnreadChannelPosts() {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const unreadChannelIds = getUnreadChannelIds(state);
+
+        const promises = [];
+        const promiseTrace = [];
+        unreadChannelIds.forEach((channelId) => {
+            const postIds = getPostIdsInChannel(state, channelId);
+
+            let promise;
+            const trace = {
+                channelId,
+                since: false,
+            };
+            if (!postIds || !postIds.length) {
+                // Get the first page of posts if it appears we haven't gotten it yet, like the webapp
+                promise = Client4.getPosts(channelId);
+            } else {
+                const since = getChannelSinceValue(state, channelId, postIds);
+                promise = Client4.getPostsSince(channelId, since);
+                trace.since = since;
+            }
+
+            promises.push(promise);
+            promiseTrace.push(trace);
+        });
+
+        let posts = [];
+        const actions = [];
+        if (promises.length) {
+            const results = await Promise.all(promises);
+            results.forEach((data, index) => {
+                const channelPosts = Object.values(data.posts);
+                if (channelPosts.length) {
+                    posts = posts.concat(channelPosts);
+
+                    const trace = promiseTrace[index];
+                    if (trace.since) {
+                        actions.push(receivedPostsSince(data, trace.channelId));
+                    } else {
+                        actions.push(receivedPostsInChannel(data, trace.channelId, true, data.prev_post_id === ''));
+                    }
+
+                    actions.push({
+                        type: ViewTypes.RECEIVED_POSTS_FOR_CHANNEL_AT_TIME,
+                        channelId: trace.channelId,
+                        time: Date.now(),
+                    });
+                }
+            });
+        }
+
+        console.log(`Fetched ${posts.length} posts from ${unreadChannelIds.length} unread channels`); //eslint-disable-line no-console
+        if (posts.length) {
+            actions.push(receivedPosts({posts}));
+            const additional = await dispatch(getPostsAdditionalDataBatch(posts));
+            if (additional.length) {
+                actions.push(...additional);
+            }
+
+            dispatch(batchActions(actions));
+        }
```

Comment:
`getPostsAdditionalDataBatch` leads me to believe that the result of this method is a set of posts, not yet another dispatchable action. Same with `receivedPosts`... seems like I should be left with a set of posts, rather than it being an action.

Once I got these clear in my head, I 100% get what's going on here now 👍 Not something for now, by any means.... this is pre-existing naming (also shared by web client) that can be cleared up once redux is ported in.